src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonValueTrimmable.cs

```diff
@@ -51,5 +51,98 @@ public override void WriteTo(Utf8JsonWriter writer, JsonSerializerOptions? optio
                 JsonSerializer.Serialize(writer, _value, _jsonTypeInfo);
             }
         }
+
+        internal override JsonNode DeepCloneValue()
+        {
+            if (_converter is not null)
+            {
+                return _value is JsonElement element
+                    ? new JsonValueTrimmable<JsonElement>(element.Clone(), JsonMetadataServices.JsonElementConverter, Options)
+                    : new JsonValueTrimmable<TValue>(_value, _converter, Options);
+            }
+            else
+            {
+                Debug.Assert(_jsonTypeInfo != null);
+                return JsonSerializer.SerializeToNode(_value, _jsonTypeInfo)!;
+            }
+        }
+
+        internal override bool DeepEquals(JsonNode? node)
+        {
+            if (node is null)
+            {
+                return false;
+            }
+
+            if (_jsonTypeInfo is not null)
+            {
+                JsonNode? jsonNode = JsonSerializer.SerializeToNode(_value, _jsonTypeInfo);
+                return DeepEquals(jsonNode, node);
+            }
+            else
+            {
+                if (node is JsonArray || node is JsonObject)
+                {
+                    return false;
+                }
+
+                if (node is JsonValueTrimmable<JsonElement> jsonElementNodeOther && _value is JsonElement jsonElementCurrent)
+                {
+                    if (jsonElementNodeOther._value.ValueKind != jsonElementCurrent.ValueKind)
+                    {
+                        return false;
+                    }
+
+                    switch (jsonElementCurrent.ValueKind)
+                    {
+                        case JsonValueKind.String:
+                            return jsonElementCurrent.GetString() == jsonElementNodeOther._value.GetString();
```

Comment:
Rather than allocating strings, I think it might be possible to use the `JsonElement.ValueEquals` method here.